### ISSUE TITLE
Maximum recursion depth exceeded

### ISSUE THREAD

**Issue by num3ri** [buckyroberts/Spider#33]
Hi all!
I tried to use the spider onto a site with about 8000 pages.
While the spider was crawling at a certain point, there was an error that stopped the execution reporting:

RuntimeError: maximum recursion depth exceeded

I corrected this error by adding, in file main.py, the statements:

import sys

sys.setrecursionlimit (100000)

**Comment by lienching** [comment]
I don't think that changing the limit of the recursion is a good idea, because that will probably cause some serious issue like Stack Overflow.

**Comment by num3ri** [comment]
any suggestions to solve the problem in another way?

**Comment by Deltatiger** [comment]
@num3ri One way I could think of is like using some sort of Queue for processing pages. That way some sort of worker pool could be created and workers can work on pages in the queue using a multi threaded approach. 
Note : This would change this from DFS (Depth First Search) ( I hope this is the case. I am yet to check out the video series fully) to BFS (Breadth First Search). 
Pros : No more recursive depth issue. If something like this is required we can also have that with some minor changes to the stuff that goes into the queue.
Cons : Very long wait times since all pages in the queue have to be processed before getting to the newly found pages.

**Comment by ShubhamWorks78** [comment]
Write the Spider again and you would not find this problem..Code has been Corrected to remove that RecursionError

**Comment by karinadelcheva** [comment]
Hey @ShubhamWorks78! Thanks for pointing that out but can you elaborate on what caused the problem in the spider? Thanks in advance
